### PR TITLE
CRM-20265, Added Card type and Pan truncation fields on Membership Form

### PR DESCRIPTION
**Pull request by pradpnayak**
----------------------------------------
* CRM-20265: Store CC type and last 4 digits from Membership form
  https://issues.civicrm.org/jira/browse/CRM-20265

CRM-20265, code cleanup and move set code into function

----------------------------------------
* CRM-20265: Store CC type and last 4 digits from Membership form
  https://issues.civicrm.org/jira/browse/CRM-20265

CRM-20265, Added CRM unit test

----------------------------------------
* CRM-20265: Store CC type and last 4 digits from Membership form
  https://issues.civicrm.org/jira/browse/CRM-20265

CRM-20265, Store Card Type and Pan Truncation for Submit Credit Card Membership
Added Unit test

----------------------------------------
* CRM-20265: Store CC type and last 4 digits from Membership form
  https://issues.civicrm.org/jira/browse/CRM-20265

CRM-20265, Added form fields in Membership Renewal Form

----------------------------------------
* CRM-20265: Store CC type and last 4 digits from Membership form
  https://issues.civicrm.org/jira/browse/CRM-20265

### CODE REVIEW

**Comment by pradpnayak** [comment]
depends on https://github.com/civicrm/civicrm-core/pull/9878

**Comment by pradpnayak** [comment]
depends on https://github.com/civicrm/civicrm-core/pull/10050

File: templates/CRM/Contribute/Form/PaymentDetails.tpl [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    }
+    else if (paymentInstrument == {/literal}{$creditVal}{literal}) {
+      $('tr#cardType').show();
+      $('tr#cardNumber').show();
+      $('tr#checkNumber').hide();
+    }
+    else {
+      $('tr#checkNumber').hide();
+      $('tr#cardType').hide();
+      $('tr#cardNumber').hide();
```
Comment (colemanw) [inline templates/CRM/Contribute/Form/PaymentDetails.tpl]: Please combine jQuery selectors e.g.

    $('tr#checkNumber, tr#cardType, tr#cardNumber').hide();

**Comment by colemanw** [comment]
Looks good. We'll re-run tests once the dependencies are merged.

**Comment by eileenmcnaughton** [comment]
Note that I am still opposed to this - I did go through adding pan_truncation to the payment form - https://github.com/civicrm/civicrm-core/pull/10157 - which would be a replacement PR - it should work on membership form if you merge that too - although you would need to check. I excluded card_type in that PR for reasons noted on that PR

**Comment by monishdeb** [comment]
@JoeMurray @pradpnayak @eileenmcnaughton I am closing this PR in favor of https://github.com/civicrm/civicrm-core/pull/10157

**Pull request closed without merging**